### PR TITLE
First Running Version of Elastic Cloud CLI

### PR DESCRIPTION
**Pull request by gibsonbailey**
This runs as I expect it to. There is an error that still occurs when paramiko attempts to ssh into a newly spun-up VM. This should not occur when the `python Cloud.py control-load gce` script is run more than 20 or 30 seconds apart though.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -102,3 +102,9 @@ venv.bak/
 
 # mypy
 .mypy_cache/
+
+# Google Cloud Service Account Information
+service_account/*
+.states
+.states/*
+scripts/GCE_CPU_Startup.sh
```
Comment (ckcollab) [inline .gitignore:110]: why not keep this in repo?

File: Cloud.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+def adapter_choice(driver):
+    if driver == 'gce':
+        return GCEAdapter()
+    else:
+        raise NotImplementedError
+    
+
+@click.group()
+def cli():
```
Comment (ckcollab) [inline Cloud.py:13]: is this necessary?

File: Cloud.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    pass
+
+@cli.command()
+@click.argument('driver', type=click.Choice(['gce']))
+def control_load(driver):
+    adapter = adapter_choice(driver)
```
Comment (ckcollab) [inline Cloud.py:19]: do we use `adapter_choice` anywhere else? prob doesn't need to be a func

File: Cloud.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+            click.echo(adapter.expand())
+
+
+@cli.command()
+@click.argument('driver', type=click.Choice(['gce']))
+def dump_state(driver):
+    adapter = adapter_choice(driver)
+
+    states = adapter.dump_state()
+    print('\n\nNode states:\n')
```
Comment (ckcollab) [inline Cloud.py:42]: switch to `echos`

File: Cloud.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    print('{0: <30} {1}'.format('Name', 'State'))
+    for state in states:
+        print('{0: <30} {1}'.format(state[0], state[1]))
```
Comment (ckcollab) [inline Cloud.py:45]: samesies

File: Cloud.py [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+@cli.command()
+@click.argument('driver', type=click.Choice(['gce']))
+def expand(driver):
+    adapter = adapter_choice(driver)
+    click.echo('Expanding...')
+    click.echo(adapter.expand())
+    
+if __name__ == '__main__':
+    cli()
```
Comment (ckcollab) [inline Cloud.py:62]: oh, I see I think?

File: ElasticCloudAdapter.py [hunk 7]
```diff
@@ -0,0 +1,61 @@
+import os
+import paramiko
+import yaml
+
+
+# TODO: When paramiko is updated > 2.4.2 remove this warning squelch
```
Comment (ckcollab) [inline ElasticCloudAdapter.py:6]: add a date to this, "last checked Mar 14 2019" or something

File: scripts/GCE_GPU_Startup_Explained [hunk 8]
```diff
@@ -0,0 +1,32 @@
+Setup GPU driver:
```
Comment (ckcollab) [inline scripts/GCE_GPU_Startup_Explained:1]: could make this an MD and format the text?

**Review by ckcollab — CHANGES_REQUESTED** [review]

File: Cloud.py (hunk 2 above)
Comment (gibsonbailey) [reply Cloud.py:13]: I believe that this is used to show all of the grouped commands when 
`python Cloud.py --help` is invoked.

File: Cloud.py (hunk 3 above)
Comment (gibsonbailey) [reply Cloud.py:19]: This is used at the beginning of every command (control-load, expand, shrink, dump-state).

File: Cloud.py (hunk 5 above)
Comment (gibsonbailey) [reply Cloud.py:45]: By samesies, do you mean that you want these changed to echos, or do you want the formatting strings in a string variable?

File: scripts/GCE_GPU_Startup_Explained (hunk 8 above)
Comment (gibsonbailey) [reply scripts/GCE_GPU_Startup_Explained:1]: Great idea, although I think we should finalize the script on your GCE account to be sure it works with GPU accelerators before I build a pretty MD explanation. If you disagree, I'll happily go through it.

File: .gitignore (hunk 1 above)
Comment (gibsonbailey) [reply .gitignore:110]: I suppose these are fine to stay in. Other than the user's service account key.

File: Cloud.py (hunk 4 above)
Comment (gibsonbailey) [reply Cloud.py:42]: Done.

File: Cloud.py (hunk 5 above)
Comment (gibsonbailey) [reply Cloud.py:45]: Either way, I did both.

File: ElasticCloudAdapter.py (hunk 7 above)
Comment (gibsonbailey) [reply ElasticCloudAdapter.py:6]: Done.

File: .gitignore (hunk 1 above)
Comment (gibsonbailey) [reply .gitignore:110]: The .states directory contains files that are created and used at runtime by Cloud.py. I suppose that the directory itself can stay, but the contents should not.

**Pull request closed without merging**